agent/base

```diff
@@ -12,24 +12,20 @@ set -o pipefail
 # very first thing to do is figure out which pbench we are
 if [[ -z "$pbench_run" ]]; then
     pbench_run=$(pbench-config pbench_run pbench-agent)
-    if [[ -z "$pbench_run" ]]; then
-        pbench_run=/var/lib/pbench-agent
-    fi
     export pbench_run
-else
-    if [[ ! -d $pbench_run ]]; then
-        echo "[ERROR] the provided pbench run directory, ${pbench_run}, does not exist" >&2
-        exit 1
-    fi
 fi
 
 # the pbench temporary directory is always relative to the $pbench_run
 # directory
 pbench_tmp="$pbench_run/tmp"
-mkdir -p $pbench_tmp
+mkdir $pbench_tmp
 if [[ ! -d $pbench_tmp ]]; then
+  if [[ ! -d $pbench_run ]]; then
+    echo "[ERROR] the provided pbench run directory, ${pbench_run}, does not exist" >&2
+  else
     echo "[ERROR] unable to create TMP dir, ${pbench_tmp}" >&2
-    exit 1
+  fi
+  exit 1
```

Comment:
I think it's a more common pattern to check `${?}` after a command to see whether it worked. In this case, since we're requiring and assuming that the `pbench_run` directory already exists, I think we'd just want something simple like

```
mkdir ${pbench_tmp}
if [[ ${?} -ne 0 ]]; then
    echo "[ERROR] unable to create TMP directory, ${pbench_tmp}" >&2
    exit 1
fi
```

Reply:
I wanted to keep it simple. but if an environment path for `pbench_run` is passed and it is not already created then this will help us to know the reason behind that. But if we wanna rely on this msg then I will remove that check :)

Reply:
Fair enough; there's nothing wrong with this, really, and it probably is good to distinguish between the two cases. Using the `-d` check instead of testing `${?}` just looks a bit odd. I suppose you could combine the two by checking `pbench_run` existence only if the `mkdir` failed, but it's probably not important.

Reply:
Ok